code/game/machinery/ai_relay.dm

```diff
@@ -0,0 +1,176 @@
+GLOBAL_LIST_EMPTY(ai_relays)
+
+#define AI_RELAY_HACK_TIME 600
+#define FLIP_DELAY 150
+
+/obj/machinery/ai_relay
+	name = "processing relay"
+	desc = "A mighty piece of hardware used to process massive amounts of data. It allows the AI to interface with various machines on the station."
+
+	icon = 'icons/obj/machines/telecomms.dmi'
+	icon_state = "hub"
+
+	critical_machine = TRUE
+	density = TRUE
+
+	use_power = IDLE_POWER_USE
+	idle_power_usage = 500
+	active_power_usage = 5000 //10x
+
+	var/on = FALSE
+
+	var/list/linked_ais = list()
+
+	var/list/hacked_ais = list()
+
+	var/module_door_open = FALSE
+	var/obj/item/ai_relay_module/inserted_module
+
+	var/last_flipped
+
+	circuit = /obj/item/circuitboard/machine/ai_relay
+
+/obj/machinery/ai_relay/Initialize()
+	GLOB.ai_relays += src
+	name += " ([num2hex(rand(1,65535), -1)])"
+	. = ..()
+
+/obj/machinery/ai_relay/Destroy()
+	GLOB.ai_relays -= src
+	reset_connected_ais()
+	. = ..()
+
+/obj/machinery/ai_relay/attack_hand(mob/user)
+	. = ..()
+	if(stat & NOPOWER)
+		to_chat(user, "<span class='warning'>Nothing happens...</span>")
+		return
+
+	if(panel_open)
+		to_chat(user, "<span class='warning'>You need to close the maintenance hatch first!</span>")
+		return
+
+	if(module_door_open && inserted_module)
+		inserted_module.forceMove(get_turf(src))
+		user.visible_message("[user] removes [inserted_module].", "<span class='notice'>You remove [inserted_module].</span>")
+		inserted_module = null
+		return
+
+	if(on)
+		if(last_flipped > world.time)
+			to_chat(user, "<span class='warning'>The breaker won't budge! The relay has locked it for [(last_flipped - world.time) / 10] seconds!</span>")
+			return
+		last_flipped = world.time + FLIP_DELAY
+		on = FALSE
+		user.visible_message("[user] switches [src] off.", "<span class='warning'>You flip the breaker to the OFF position.</span>")
+		use_power = IDLE_POWER_USE
+		message_ais("<span class='warning'>[src] has been turned manually off!</span>")
+		update_icon()
+	else
+		if(last_flipped > world.time)
+			to_chat(user, "<span class='warning'>The breaker won't budge! The relay has locked it for [(last_flipped - world.time) / 10] seconds!</span>")
+			return
+		last_flipped = world.time + FLIP_DELAY
+		on = TRUE
+		user.visible_message("[user] switches [src] on.", "<span class='warning'>You flip the breaker to the ON position.</span>")
+		use_power = ACTIVE_POWER_USE
+		message_ais("<span class='notice'>[src] has been turned manually on.</span>")
+		update_icon()
+
+
+/obj/machinery/ai_relay/process()
+	if(stat & (NOPOWER|BROKEN))
+		visible_message("[src] switches off.")
+		update_icon()
+		on = FALSE
+		use_power = IDLE_POWER_USE
+		return FALSE
```

Comment:
```suggestion
		return PROCESS_KILL
```
Then re-start the process when you need it?

Reply:
Nah we would need to run a process or similar  to detect the power change

Reply:
I guess that works

Reply:
What I’m saying is that it’s the same either way